frontend/exporter/src/body.rs

```diff
@@ -95,8 +95,16 @@ mod module {
         impl<MirKind: IsMirKind + Clone + 'static> IsBody for MirBody<MirKind> {
             fn body<'tcx, S: UnderOwnerState<'tcx>>(did: RLocalDefId, s: &S) -> Self {
                 let (thir, _) = get_thir(did, s);
-                let mir = Rc::new(s.base().tcx.mir_built(did).borrow().clone());
-                mir.sinto(&with_owner_id(s.base(), thir, mir.clone(), did.to_def_id()))
+                let mir = MirKind::get_mir(s.base().tcx, did, |body| {
+                    let body = Rc::new(body.clone());
+                    body.sinto(&with_owner_id(
+                        s.base(),
+                        thir,
+                        body.clone(),
+                        did.to_def_id(),
+                    ))
+                });
+                mir.unwrap()
```

Comment:
Can you use `s_unwrap` here? If the unwrap fails, that will show a spanned error.
(i.e. `mir.s_unwrap(s)`)